### ISSUE TITLE
Add options to hide title and status bar

### ISSUE THREAD

**Issue by WindSoilder** [zed-industries/zed#5120]
### Description

Usually I feel some components of zed editor(title and status bar) are redundant to me, and I just want to focus on the editor itself.  It's something called zen mode in vscode.

It's something similar to zed-industries/zed#5333, but I want to hide more..

### Describe the solution you'd like

Here is the components I want to be able to hide:
![image](https://github.com/user-attachments/assets/ae4f6777-12bd-4771-9b12-4036579d6c3d)

Here is how vscode shows:
![image](https://github.com/user-attachments/assets/57cf782b-1e86-405a-ba69-42bda5b070b5)

**Comment by stjepangolemac** [comment]
This would be fantastic along with 
- https://github.com/zed-industries/zed/issues/4685 🚀

**Comment by DissolveDZ** [comment]
still waiting for this to happen, along with being able to hide the tabs

**Comment by philippbosch** [comment]
@rsmirnov90 you can hide the breadcrumb bar like this: 

```json
{
  // …
  "toolbar": {
    "breadcrumbs": false,
    "quick_actions": false
  },
```

**Comment by mcpeng** [comment]
We have Titlebar, TabBar, Toolbar, StatusBar, they occupy too much valuable vertical space, like old M$ Office. Considering the characteristics of collaboration, perhaps they can be placed in the StatusBar.

**Comment by yasinkavakliat** [comment]
Guys, it's 2024 and we don't have an option to disable the status bar? That should've been added in the first beta.
Come on, I can't switch to Zed when the UI is full of noise. I want to disable everything that I don't need (includes the sign in button, status bar, tabs, titlebar and everything else).

**Comment by mhanuszh** [comment]
When (if?) we get this feature, can we have the ability to still show the panel buttons, just like in VSCode?

My current layout:
![vscode](https://github.com/zed-industries/zed/assets/5262962/729764fe-5ded-4001-aa8e-74c7f6098207)

**Comment by themahdavi** [comment]
options for full customizing UI are one of basic things on editors. please add soon as possible.
we don't need second visual studio code just add basics
Thanks.

**Comment by dekdevy** [comment]
Big bump. Onboarding zed has been a breeze but when I see that top title bar I am seriously wondering what the thought process was. It does nothing. Same for the status bar height. Editors keep getting these absolute basics wrong and I dont understand why.  Please let users configure what happens on their screen real estate.

- Top Window bar height / disable entirely
- Status bar height / disable entirely
- Project Panel: vertical line height. Also, default indent seems insanely large - at can be edited, so its fine!

A permanent "sign in" button isnt giving off the right vibes - I think many people are looking for something in between vscode and sublime, and initially zed seems like a great tool in the right direction, but when it throws "sign in" "use github" "use assistant" "use chat" "collaborate with team members" "use copilot" your way when opening for the first time, I feel like im headed right back into windows 11.

**Comment by qqap** [comment]
A bit of a temporary workaround, but to completely hide the status bar, and blank the title bar while maintaining a functional code editor and terminal, you can just change the `ui_font_size` to 0.

```
  "ui_font_size": 0,
  "buffer_font_size": 14,
```

**Comment by dlindenkreuz** [comment]
For me, the toolbar ("title bar") is hidden when adding these settings:

```json
"toolbar": {
  "breadcrumbs": false,
  "quick_actions": false
},
```

**Comment by JulienLecoq** [comment]
> For me, the toolbar ("title bar") is hidden when adding these settings:
> 
> ```json
> "toolbar": {
>   "breadcrumbs": false,
>   "quick_actions": false
> },
> ```

Weird, I still have it with this (on macOS).

**Comment by juliangarnier** [comment]
> > For me, the toolbar ("title bar") is hidden when adding these settings:
> > ```json
> > "toolbar": {
> >   "breadcrumbs": false,
> >   "quick_actions": false
> > },
> > ```
> 
> Weird, I still have it with this (on macOS).

Doesn't work for me either.

**Comment by juliangarnier** [comment]
@iamnbutler I understand based on #12325 that removing the Sign in button is not an option, but would you at least consider hiding the title bar in full screen mode?

I really appreciate the care and attention that have been put into the UI, Zed feels extremely polished, but it's also the only editor that I tried that won't let me take advantage of most of the screen space when entering full screen mode.

In fullscreen, it feels like browsing a website that won't let you close a sticky newsletter signup banner:

![sticky-title](https://github.com/user-attachments/assets/9718f4a3-dd37-4722-8535-7b9518c510d1)

When you decide to go fullscreen, you want the editor UI to disappear as much as possible. Having the title bar always visible is the reason why I keep going back to Sublime.

I think a lof of issues could be solved with a more minimal fullscreen mode (#13218, #4382, #6862, #12325, #9730) and won't have to impact the current UI when used in normal "windowed" mode.

Thanks for considering, Zed is really great!

**Comment by AndydeCleyre** [comment]
@dlindenkreuz

> For me, the toolbar ("title bar") is hidden when adding these settings:

Can you verify which bar you mean? For me, those settings hide the *toolbar* (below tabs), and I appreciate that, but I'm really looking to get rid of what I think Zed calls its *title bar* (above tabs). 

The title bar has the following items, none of which I want to see:

- application menu
- recent folders button
- branch button (I wouldn't mind it in the status bar)
- sign in button
- user menu

**Comment by dlindenkreuz** [comment]
<img width="839" alt="Screenshot 2024-10-24 at 17 31 35" src="https://github.com/user-attachments/assets/b83a0c03-77bd-4ba4-9767-c67af7887818">

This is what it looks like for me. The title bar above tabs is not hidden. Sorry, I must have misread the question.

**Comment by jon-walstedt** [comment]
I agree with @juliangarnier, having an option to hide the top bar entirely in fullscreen mode would be great, if the only reason is the User Menu toggle I would suggest moving it to the bottom status bar somewhere instead. To me the top bar is a total waste of space right now and makes the editor look weird in full screen mode.

<img width="1728" alt="Screenshot 2024-12-18 at 21 30 56" src="https://github.com/user-attachments/assets/63656049-39e7-4252-9097-95a47290e656" />

**Comment by mcpeng** [comment]
In fullscreen mode, it is necessary to have an option to hide the top bar. Perhaps it could work like Safari or the latest version of Chrome, where the toolbar only appears automatically when the cursor moves to the top (disable “View > Always Show Toolbar in Fullscreen”).

**Comment by dekdevy** [comment]
This has been widely requested and not been acknowledged, so I'm guessing we're not going to get it. I may just fork the thing to implement a bunch of low hanging fruit screen real-estate fixes

**Comment by m4saurabh** [comment]
Do it please it just bugs every time I notice it, it's especially a problem on smaller non-retina screen where you don't get enough screen real-estate to begin with and text need to be large otherwise it look like unreadable garbage.

**Comment by pawurb** [comment]
I work with scaled-up resolution, and bottom bar is obscuring a lot of workspace. I'd love to get a feature to hide it entirely.

**Comment by mcpeng** [comment]
Maybe there is different version?
>https://zed.dev/blog/edit-prediction

<img width="673" alt="Image" src="https://github.com/user-attachments/assets/8017b5dd-3b11-40de-beaa-bdb472bcc61d" />

**Comment by mcobzarenco** [comment]
> I agree with @juliangarnier, having an option to hide the top bar entirely in fullscreen mode would be great, if the only reason is the User Menu toggle I would suggest moving it to the bottom status bar somewhere instead. To me the top bar is a total waste of space right now and makes the editor look weird in full screen mode.

Not just in full screen mode, I would like the option to have no window decorations whatsoever at all, they're pointless when not using a mouse. E.g. in my Emacs config:

![Image](https://github.com/user-attachments/assets/beffd51d-3a91-4231-8286-6875467a307d)

**Comment by an-h2** [comment]
Just noticed we had a previous PR for this, but it was closed. Would anyone like to make a fork for temporary use? I'm afraid I don't know Rust well enough to make it work.

https://github.com/zed-industries/zed/pull/11137

**Comment by x70b1** [comment]
I have been following this issue for a while now. I don't think zed is seriously interested in it.

zed's only monetization opportunity is collaboration. So the login is an important thing for this. They want to annoy you with it. If you hide the login, you escape zed's monetization option.

zed has no interest in you using a simple editor without a login button. zed has no interest in you using an editor that you are not logged in to. Zed has no interest in you using a good editor without having paid for it.

Thats the reason why the bar and the button is there. You can move around a lot of things at the bottom bar for example. But the top bar is untouchable. Therefore I dont think that we will see a change here in the near future. 

I don't want to say that monetization is bad. Everyone has to earn money. Even open source is not build at zero cost. Someone has worked for it.

**Comment by mcobzarenco** [comment]
Wouldn't they accept a PR which implements this feature behind an option though? Even if it's not a priority for them to implement. The amount of visual real estate used for (arguably irrelevant) stuff in the default configuration of zed stood out to me as a first impression.

Just in case someone from zed reads this, I would be happy to pay a monthly subscription for a high quality oss editor which allows me to move away from emacs. The login stuff and needing to be connected to the network (e.g. even when developing a local extension) is not ideal.

**Comment by PedroHLC** [comment]
Their last statement on the subject still is: https://github.com/zed-industries/zed/pull/25879#pullrequestreview-2652719621. It would be useful to read it before commenting on the issue onward.

**Comment by dekdevy** [comment]
> Their last statement on the subject still is: [#25879 (review)](https://github.com/zed-industries/zed/pull/25879#pullrequestreview-2652719621). It would be useful to read it before commenting on the issue onward.

I read these comments and I don't understand the reasoning. Please correct me if you think I am misreading this, but it appears as "hiding this stuff is extremely complicated, because we need to be able to show things to you". 

I want to hide the information. The user intent should be clear once they explicitly opt into hiding the information. I'll be okay.

Also, I'd be happy to throw good money at a tool like zed if it presented itself as something you own forever and can customize more. Sublime was a no-brainer purchase for me back then. I feel that right now the design process is hinting at a slow degradation into Adobe-like philosophy. This is unfortunate. It's not like I wouldn't consider signing up to a monthly payment for LLM tokens just because I purchase the editor.

**Comment by up-to-you** [comment]
Any monetization can be done via notifications pushing them up from any corner of the screen or even via dialogs, so I don't think monetization is the reason.
What's really hilarious is that i spent almost a day to shift all my settings/hotkeys/somewhere the workarounds etc. from the WebStorm and Clion and have found this issue only at the end of my process...

It's cooked, welcome to the new opinioned GNOME-like product.

**Comment by TuralAsgar** [comment]
It would be great to be able to hide the status bar.
In JetBrains editors, it is possible.

**Comment by gdkrmr** [comment]
I would love to have a more configurable status bar. All the information from the tool bar and top bar could be shown in the  status bar. This frees up quit a bit of screen real estate and unclutters the UI.

**Comment by bbkingisking** [comment]
My main problem was the persistent "Sign-in" button, so I made a fork and I hardcoded it out https://github.com/bbkingisking/zed-no-sign-in
2 lines of diff

**Comment by up-to-you** [comment]
@bbkingisking can you also hardcode out status and title bars, please ?

**Comment by bbkingisking** [comment]
@up-to-you I could try. Fork the main repo and I'll submit a PR

**Comment by bbkingisking** [comment]
@mcpeng yep that was useful, I've added it to the fork. Now 	"show_status_bar": false hides the status bar.

**Comment by asherqnovick** [comment]
It's hard to understand why this simple feature still doesn't exist after 2+ years. This is so close to being a perfect product that I will happily pay for but the inability to make simple and very standard UI customizations to a text editor makes me second guess using this product at all. 

Do these developers not use their own product every day and get annoyed with the massive amount of blank space at the top and bottom of the screen? I guess they think we're more likely to "Sign In" if we see the sign in button every day, but don't realize I won't be seeing the sign in button at all if I'm not using your product at all...specifically for the reason that there's a sign in button in your text editor that I can't make go away.

You understand your userbase enough to natively implement vim motions but you don't allow us to actually fullscreen the editor....baffling.

**Comment by kruegernet** [comment]
> It's hard to understand why this simple feature still doesn't exist after 2+ years. This is so close to being a perfect product that I will happily pay for but the inability to make simple and very standard UI customizations to a text editor makes me second guess using this product at all.
> 
> Do these developers not use their own product every day and get annoyed with the massive amount of blank space at the top and bottom of the screen? I guess they think we're more likely to "Sign In" if we see the sign in button every day, but don't realize I won't be seeing the sign in button at all if I'm not using your product at all...specifically for the reason that there's a sign in button in your text editor that I can't make go away.
> 
> You understand your userbase enough to natively implement vim motions but you don't allow us to actually fullscreen the editor....baffling.

It's a product to make money, not a tool for people to use :(

people like you and I are probably better off with avante.nvim etc

**Comment by rrveex** [comment]
Thank you zed-industries for this wonderful **free** **open-source** product!

I'll never be interested in logging in, regardless how often I'll be seeing the button, so I took the liberty to change **2 lines of code** in order to hide it. It took me 10 minutes to find them (and a bit longer to build) and I'm no Rustafarian.

![Image](https://github.com/user-attachments/assets/7cfa2b54-5a95-485d-8d41-6c690a1458dc)

P.S. I like the status bar, so I left that as it was.

**Comment by P1n3appl3** [comment]
For what it's worth, you can already disable the sign-in button in zed and user portrait with the following settings:

```json
"title_bar": {
  "show_sign_in": false,
  "show_user_picture": false
}
```

**Comment by rrveex** [comment]
Yes, I don't think anybody's bothered by the login button, rather the wasted space by the whole bar. I should have formulated differently.

**Comment by pawanmkr** [comment]
Yes, We want to get rid of this...Please check the below screenshot, that horizontal bar below tab bar is also annoying and can be hidden

<img width="1136" alt="Image" src="https://github.com/user-attachments/assets/f9b33a0d-4444-47e2-9672-57d71b60fb7a" />

**Comment by akyoto** [comment]
Even the infamous VS Code is more configurable than Zed.

![Image](https://github.com/user-attachments/assets/daa07770-9b32-4453-8e8f-9060bdba9746)

I don't want to sound too negative but this has been a blocker for adoption of Zed since I've heard about it.
Can we maybe get a clarification on the team's stance towards UI configuration, especially in regard to minimalist UIs?

Like dekdevy already said, we don't need to be babysit about potentially missed notifications or features.
The user **intentionally** decides to hide these elements. It's **okay**.

**Comment by somtooo** [comment]
> It's hard to understand why this simple feature still doesn't exist after 2+ years. This is so close to being a perfect product that I will happily pay for but the inability to make simple and very standard UI customizations to a text editor makes me second guess using this product at all.
> 
> Do these developers not use their own product every day and get annoyed with the massive amount of blank space at the top and bottom of the screen? I guess they think we're more likely to "Sign In" if we see the sign in button every day, but don't realize I won't be seeing the sign in button at all if I'm not using your product at all...specifically for the reason that there's a sign in button in your text editor that I can't make go away.
> 
> You understand your userbase enough to natively implement vim motions but you don't allow us to actually fullscreen the editor....baffling.

what a crazy reply to a free and opensource project if status bar is causing you grief open a pr. nonsense

**Comment by negentropi** [comment]
yeah, vsc is more configable. zed static defult lock screen space for gui.

file history/timeline the same.

really love the zed fix run text editor inside browser eng ; used outside web n offline. just didnot solve better defult.

**Comment by huangmiuXyz** [comment]
i need hide title_bar

**Comment by bgub** [comment]
Just submitted a PR to address the rest of this issue, building on the work by @aabiji :)

https://github.com/zed-industries/zed/pull/37428

**Comment by SomeoneToIgnore** [comment]
Back to 0 with https://github.com/zed-industries/zed/pull/37428#issuecomment-3326395830

**Comment by dekdevy** [comment]
..wow. I was getting excited.

A tiny but frequently requested PR being blocked for years - now implying that there will be something in that space that we *must* look at - despite the fact that anyone requesting this feature is explicitly asking for *the UI not using screen real estate*.   

This was accepted, shipped in preview 205, then later rejected. The devs were on board. These kind of anti-pattern "yeah, you definitely need this aggressively annoying button on your UI because it helps stock prices" is how adobe lost the sentiment.

Please guys, dont fall into the adobe trap.

<img width="550" height="330" alt="Image" src="https://github.com/user-attachments/assets/834d616d-555f-4e03-8410-f707b2e9e5c3" />

**Comment by BigMitchGit** [comment]
> [@rsmirnov90](https://github.com/rsmirnov90) you can hide the breadcrumb bar like this:
> 
> {
>   "toolbar": {
>     "breadcrumbs": false,
>     "quick_actions": false
>   },
> }

Oh great, just did that. Would've been more clear and consistent with a "show" key. It would be really nice if the breadcrumbs could be added to the bottom bar. Because I do like the breadcrumbs but dont want them to take up extra screen real estate, and the bottom bar is already there with lots of free space.

**Comment by boozook** [comment]
> [@rsmirnov90](https://github.com/rsmirnov90) you can hide the breadcrumb bar like this:
> ```json
> {
>   "toolbar": {
>     "breadcrumbs": false,
>     "quick_actions": false
>   },
> }
> ```

Unfortunately this is useless if you need to remove unnecessary bar entirely for vertical space economy and decrease disturbance by unnecessary things like collab - not all users need and use collab for example, as well as api-connection status that I believe must be removable and should be in the bottom status bar.

**Comment by BigMitchGit** [comment]
@boozook that does remove that toolbar completely. it makes the entire toolbar invisible.

**Comment by boozook** [comment]
@BigMitchGit definitely nope, because toolbar is not titlebar. I don't need __titlebar__ as well as toolbar, but it's more critical then toolbar. Okay, that was my mistake, sorry, seems to we still have `title_bar.show = "hide_in_full_screen"` in the `v0.206.6`.

**Comment by KorigamiK** [comment]
this issue can be marked closed!

**Comment by m4saurabh** [comment]
nah, the status bar is still remaining.

**Comment by wweir** [comment]
A shorter height would suffice.

**Comment by aeimi** [comment]
> nah, the status bar is still remaining.

All I want to have is tabs to the top edge of screen. Like Chrome browser. So easier to navigate them with mouse. Sublime Text 4 do this, but only in full screen mode. And when title is hidden there is no file path below tabs. If Zed finally had it, it would be great. On Linux and Windows.

**Comment by xipeng-jin** [comment]
Hello Zed, I would like to share some adjustments for hiding title bar with tab bar potentially in my own [fork](https://github.com/xipeng-jin/zed/tree/feature/minimal-ui). Since I am using a 14-inch MacBook for my daily driver and I love to use Zed editor so much ❤️, but the only thing annoying me is the large top title bar occupied lots of space in my vertical real-estate as people mentioned in this thread. And personally I rarely touch any buttons in the title bar. Here is the screen recording for my version of implementation:

https://github.com/user-attachments/assets/6751d9ee-3c27-4967-a43f-2c9da3748f35

Here is the list for the new UI settings capability
1. Introduces a `title_bar.show` setting wired through the Title Bar module to dynamically attach or remove the native chrome.
2. Extends the windowing layer with APIs to vertically center macOS traffic lights and to toggle their visibility, letting the chrome float precisely where the condensed layout needs it when the title bar disappears.
3. Rethinks the left dock so it can reserve space for those controls, adds a native-height placeholder, and keeps the dock layout stable as panes open and close—no more traffic-light overlap in minimal mode as mentioned in this issue #38547 .
4. Links the new dock reservation to runtime state: when the title bar is hidden the dock now owns the traffic lights, shows them only when it’s open, and keeps them anchored to the tab stack.
5. Tightens the workspace toolbar with smaller padding, adaptive borders, and awareness of the tab bar’s visibility, trimming visual chrome so editors and terminals get every available vertical line.

And I think we already have an official supported experimental feature to also hide status bar completely by setting:
```
"status_bar": {
  "experimental.show": false
}
```
So now we should have ability to hide the title bar, tab bar, and status bar at the same time which means we are approaching to accomplish the [zen mode](https://github.com/zed-industries/zed/discussions/36882) in Zed 🎉.

Finally, I really appreciate this PR #37428 to give me some initial inspirations for doing my own work on top of it. The reason why I didn't open a PR for it because I saw some comments in that thread about 

> there are plans for the new redesign to heavily use the title bar and the new redesign wants it to always be present

So I don't think merging my code for now is necessary for Zed roadmap. But I am happy to hear any feedback on it and it would be super great to inspire the future Zed UI design roadmap ☺️. Also I think it would be helpful if we are planning to develop a zen mode on top of it. Thanks for your great work Zed team!

**Comment by visualglitch91** [comment]
This was working and got reverted? Title bar makes no sense for people using tiling window managers, is there a plan to bring this back?

**Comment by stefvista** [comment]
I would really like to have the ability to hide the titlebar and I don't mind having the ability to hide those traffic lights as well as I never use them.  Currently I am sticking with alternatives purely because I find it annoying that there is unused real estate, which adds to the already unused space because of the notch on my macbook. If this is resolved I would be really happy.

**Comment by xipeng-jin** [comment]
The previous PR https://github.com/zed-industries/zed/pull/37428 got reverted because of macOS traffic lights issue in the previous version and there are new plans around redesign the title bar which makes it a must need. My current implementation has fixed the traffic light issue, but I mentioned above I am not sure opening another PR again for this is necessary for Zed roadmap.
I fully respect the plan from Zed team, and I currently build from source on my [fork](https://github.com/xipeng-jin/zed/tree/feature/minimal-ui) to fit my need for my development. I will keep my fork up to date based on the latest preview version. But at the end of day, it would be great to hear any concrete plans from Zed team so we can move forward.

**Comment by xipeng-jin** [comment]
I recently realized that the title bar might be an essential to show when you are using collab feature, since you need to render some buttons (like mic, speaker, etc..) and the list of people avatar on the title bar to be able to control during collaboration. However, those features are never turned on for the solo mode or even offline mode. Therefore I think the better way to handle it is that we would have a shortcut/keybinding to trigger the Action to toggle the title bar. People can turn it on/off as needed. 

Or if the team needs to redesign the title bar to make it even better and important, we can still have something like a Zen mode to hide any bar we like for this specific mode (My initial thought is we can tied some UI customization settings with Zen mode, so when you trigger the Zen mode the presets will be automatically enabled) and we can trigger it as needed.

Since I saw lots of people want to hide the title bar, I would like to implement this shortcut function to open another PR in the near future to get it another shot.

**Comment by xipeng-jin** [comment]
I got a new PR [#42540](https://github.com/zed-industries/zed/pull/42540) out for adding the option `title_bar.show` in the global setting and a Action `zed::ToggleTitleBar` to toggling whether or not to show the title bar in runtime. It makes sure we won't miss any functionalities from the title bar by toggling its state as needed in real time. The macOS traffic lights compatibility has been fixed in the new PR, so it should be safe to use for all platform. But I would like to have any feedback from people coming from other platforms since I only have a Macbook to test. Please reference the details in PR description. I wish Zed team can bring this feature back in this time. Thank you!

**Comment by ShalokShalom** [comment]
I like to introduce the idea of the "show tabbar only if more than one tab" 

Is this already discussed in this thread, and I simply didn't see it?

**Comment by ammi1378** [comment]
I literally have nothing in title bar

<img width="2922" height="1118" alt="Image" src="https://github.com/user-attachments/assets/e2399f83-23bb-462e-a1af-75ccaba08918" />
i would appreciate to be able to nuke that part out of existing.

**Comment by jetpham** [comment]
This is their current stance rn:
https://github.com/zed-industries/zed/pull/42540#issuecomment-3522624069

**Comment by zxsleebu** [comment]
+1. would love to see this actually huge vertical screen estate expansion

**Comment by nemeott** [comment]
> +1. would love to see this actually huge vertical screen estate expansion

Agreed. This would be very useful for those using tiling window managers. An extra 5 lines of code on the screen goes a long way.

**Comment by morgankrey** [comment]
Hi folks, closing this in favor of a discussion - this is a feature request and not a bug/issue.
